kotlinx-coroutines-core/common/src/internal/DispatchedTask.kt

```diff
@@ -22,7 +48,10 @@ internal abstract class DispatchedTask<in T>(
 
     internal abstract fun takeState(): Any?
 
-    internal open fun cancelResult(state: Any?, cause: Throwable) {}
+    /**
+     * Called when this task was cancelled while it was being dispatched.
+     */
+    internal open fun cancelCompletedResult(takenState: Any?, cause: Throwable) {}
```

Comment:
Need to figure out if `cancelCompletedResult` can throw an exception (and caller must catch) or it takes care about all the 3rd-party exceptions itself.

Reply:
Figured out that it always takes care of them. Added tests.